modules/Xsens_module/src/XsensRetargeting.cpp

```diff
@@ -0,0 +1,287 @@
+
+    for (size_t i = 0; i < m_humanJointsListName.size(); i++)
+    {
+        if (i < m_robotJointsListNames.size())
+            yInfo() << "(" << i << "): " << m_humanJointsListName[i] << " , "
+                    << m_robotJointsListNames[i];
+        else
+        {
+            yInfo() << "(" << i << "): " << m_humanJointsListName[i] << " , --";
+        }
+    }
+
+    std::string portName;
+    if (!YarpHelper::getStringFromSearchable(rf, "wholeBodyJointsPort", portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to get a string from a searchable";
+        return false;
+    }
+    if (!m_wholeBodyHumanJointsPort.open("/" + getName() + portName))
+    {
+        yError() << "[XsensRetargeting::configure] " << portName << " port already open.";
+        return false;
+    }
+
+    if (!YarpHelper::getStringFromSearchable(rf, "controllerPort", portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to get a string from a searchable";
+        return false;
+    }
+    if (!m_wholeBodyHumanSmoothedJointsPort.open("/" + getName() + portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to open the port " << portName;
+        return false;
+    }
+
+    // I should do a maping between two vectors here.
+
+    bool foundMatch = false;
+    for (unsigned i = 0; i < m_robotJointsListNames.size(); i++)
+    {
+        for (unsigned j = 0; j < m_humanJointsListName.size(); j++)
+        {
+
+            if (m_robotJointsListNames[i] == m_humanJointsListName[j])
+            {
+                foundMatch = true;
+                m_humanToRobotMap.push_back(j);
+                break;
+            }
+        }
+        if (!foundMatch)
+        {
+            yError() << "not found match for: " << m_robotJointsListNames[i] << " , " << i;
+            ;
+            return false;
+        }
+        foundMatch = false;
+    }
+
+    yInfo() << "*** mapped joint names: ****";
+    for (size_t i = 0; i < m_robotJointsListNames.size(); i++)
+    {
+        yInfo() << "(" << i << ", " << m_humanToRobotMap[i] << "): " << m_robotJointsListNames[i]
+                << " , " << m_humanJointsListName[(m_humanToRobotMap[i])];
+    }
+
+    //     m_tick =std::chrono::duration_cast< std::chrono::milliseconds
+    //     >(std::chrono::system_clock::now().time_since_epoch()); m_tock
+    //     =std::chrono::duration_cast< std::chrono::milliseconds
+    //     >(std::chrono::system_clock::now().time_since_epoch());
+    m_firstIteration = true;
+
+    double jointThreshold;
+    if (!YarpHelper::getDoubleFromSearchable(rf, "jointDifferenceThreshold", jointThreshold))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to find the whole body joint difference "
+                    "threshold.";
+        return false;
+    }
+    m_jointDiffThreshold = jointThreshold;
+
+    yInfo() << " Sampling time  : " << m_dT;
+    yInfo() << " Smoothing time : " << smoothingTime;
```

Comment:
Here you can add the suffix `[XsensRetargeting::configure]`

Reply:
done.